src/devices/src/virtio/block/device.rs

```diff
@@ -344,17 +333,21 @@ impl Block {
                     head.index, e
                 )
             });
-            used_any = true;
-        }
 
-        if !used_any {
-            METRICS.block.no_avail_buffer.inc();
+            queue.set_avail_event(mem);
+            needs_notification |= queue.needs_notification(mem, has_notification_suppression);
         }
 
-        used_any
+        // We try to batch requests and then send only 1 notification to the guest.
```

Comment:
The purpose of the suppression mechanism is to reduce the overhead. The approach here basically does something similar on top it. Can this have any negative side effects like increased latency or stalls ?